src/components/EnvironmentSelection.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Alert } from 'react-native';
+import { SegmentedButtons, Text } from 'react-native-paper';
+import { createStyles } from './BrandConfigProvider';
+import { useAsyncStorage } from '../hooks/useAsyncStorage';
+import { View } from 'react-native';
+import { emitEnvironmentToggled, useStyles } from 'src/hooks';
+
+/*
+    Debug menu for selecting a secondary environment during runtime.
+    See AppDemo.tsx for implementation. Due to caching the app will require a manual
+    reload (via metro) after switching.
+*/
+
+export function EnvironmentSelection() {
+  const [storedEnvironment, setStoredEnvironment] =
+    useAsyncStorage('environment-toggle');
+  const [selectedButton, setSelectedButton] = useState('primary');
+  const { styles } = useStyles(defaultStyles);
+
+  const selectEnvironment = useCallback(
+    (value: string) => {
+      setStoredEnvironment(value);
+      setSelectedButton(value);
+      Alert.alert(
+        `Switched to ${value} environment`,
+        'To ensure new content is loaded perform a manual refresh of the app now.',
+      );
+    },
+    [setStoredEnvironment],
+  );
+
+  // Populate selection from storage
+  useEffect(() => {
+    if (storedEnvironment.data) {
+      setSelectedButton(storedEnvironment.data);
+    }
+  }, [storedEnvironment.data]);
+
+  // Communicate updated selection to listener
+  // in useEnvironmentSelection hook
+  useEffect(() => {
+    emitEnvironmentToggled(selectedButton);
```

Comment:
One downside of this is that if this component isn't loaded the environment toggle could be out of sync.

Reply:
I think this is desired behavior, correct me if I am wrong. If the toggle component doesn't load, i.e, we are not running in development mode, we only want to use the primary environment.

Reply:
Ohhh, right. You've got an excellent point there. I have to look closer at that case.

Reply:
Merging as-is for now and will update for this case in another PR.